src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/roundabout/ComplexRoundabout.java

```diff
@@ -0,0 +1,300 @@
+package org.openstreetmap.atlas.geography.atlas.items.complex.roundabout;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.Route;
+import org.openstreetmap.atlas.geography.atlas.items.complex.ComplexEntity;
+import org.openstreetmap.atlas.geography.atlas.walker.SimpleEdgeWalker;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.ISOCountryTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+
+/**
+ * A {@link ComplexEntity} representation of a roundabout. To form a valid {@link ComplexRoundabout}
+ * the source {@link Edge} and all connected Edges tagged as a roundabout must meet the following
+ * criteria: be one way; be car navigable; together form a single closed {@link Route} that has the
+ * appropriate directionality based on the country. Adapted from the MalformedRoundaboutCheck in
+ * Atlas-Checks.
+ *
+ * @author bbreithaupt
+ * @author savannahostrowski
+ */
+public class ComplexRoundabout extends ComplexEntity
+{
+    protected static final String WRONG_WAY_INVALIDATION = "This roundabout is going the wrong direction, or has been improperly tagged as a roundabout.";
+    protected static final String INCOMPLETE_ROUTE_INVALIDATION = "This roundabout does not form a single, one-way, complete, car navigable route.";
+    private static final String EXCEPTION_MESSAGE = "Exception thrown while trying to build a ComplexRoundabout";
+    private static final List<String> LEFT_DRIVING_COUNTRIES_DEFAULT = Arrays.asList("AIA", "ATG",
```

Comment:
The source is https://en.wikipedia.org/wiki/List_of_countries_with_left-hand_traffic.